mlir/include/mlir/Dialect/Linalg/TransformOps/LinalgTransformOps.td

```diff
@@ -1021,8 +1023,7 @@ def PadOp : Op<Transform_Dialect, "structured.pad",
                       TransformHandleTypeInterface:$pad,
                       TransformHandleTypeInterface:$copy);
 
-  let assemblyFormat =
-    "$target attr-dict `:` functional-type(operands, results)";
+  let hasCustomAssemblyFormat = 1;
```

Comment:
Do we really need a custom assembly format. The IR changes dont seem to need this.

Reply:
I couldn't figure out how to avoid it.  I was following what was done for `TileUsingForOp` (which I also did for `PadOp` in a previous patch).  I welcome suggestions.

Reply:
actually i remember now. using the available directives would result in having the param/handle types inlined into the list they are passed to, e.g.
```
    %1:2 = transform.structured.tile_using_forall %0 tile_sizes [%sz : !transform.any_op, 20]
           : (!transform.any_op) -> (!transform.any_op, !transform.any_op)
```
opposed to
```
    %1:2 = transform.structured.tile_using_forall %0 tile_sizes [%sz, 20]
           : (!transform.any_op, !transform.any_op) -> (!transform.any_op, !transform.any_op)
```
like how `tile_using_for` does it. i prefered the latter and followed it for both the pad and vectorize op

Reply:
I actually wouldn't be opposed to using the dynamic list directive in favor of simpler code. even if I do personally think the resulting IR is uglier, but obviously will put personal preferences aside. Would like to hear what others think.

Reply:
I dont know about the transform ops, but the ops that implement the `ViewLikeInterface` use the custom parser/printer hooks for handling this (https://github.com/llvm/llvm-project/blob/b86e0992bfa6c58be077d82d824016f590ac5d90/mlir/include/mlir/Interfaces/ViewLikeInterface.h#L107) . Maybe that could be refactored a little bit to allow using that without needing the op to implement the interface. AFAIK, those dont have this problem.

Reply:
> I actually wouldn't be opposed to using the dynamic list directive for this in favor of simpler code. even if I do personally think the resulting IR is uglier, but obviously will put personal preferences aside. But I think there should be consistency. So I would also suggest changing tile_using_for and vectorize if possible. Would like to hear what others think.

I strongly support consistency across transform ops. Going out of our way to add syntactic consistency with only tangentially-related interface is a waste of cycles. Take a look at https://github.com/llvm/llvm-project/blob/7dfb6f571c8090803a32b5611e519b77ae74de30/mlir/include/mlir/Dialect/Linalg/TransformOps/LinalgTransformOps.td#L2004-L2008, if _that_ can be reused here, let's do that. Otherwise, let's stick with custom syntax and eventually upgrade all syntax everywhere.

Reply:
yah that's the one i was talking about. maybe i was using it wrong. let me try again

Reply:
oh, so i'm an idiot and was just using it wrong. fixed now. thanks!

Reply:
i would like to do the same for the other ops. i'll submit another pr for that

Reply:
Thanks!

Reply:
so now `vectorize`, `tile_using_for` and `pad_op` have the same syntax, where the dynamic types are in the functional-type. the `tile_using_forall` op still has the types inlined in the dynamic list. so i guess we should decide, for consistency, which one to go with. and i'll go ahead and make a PR to update all of them.

Reply:
although it looks like `PackedOrDynamicIndexList` doesn't let the types be optional like `DynamicIndexList` does

Reply:
fyi, i started updating to make consistent. but i think `PackedOrDynamicIndexList` will need modifiying in order to get `tile_using_forall` to work the same. https://github.com/srcarroll/llvm-project/compare/pad-op-parameter...srcarroll:llvm-project:consistent-transform-syntax

Reply:
I'm fine changing things to get more consistency.